src/main/scala/jio/package.scala

```diff
@@ -74,7 +64,37 @@ package object jio extends JioFiles with DecorateAsScala with DecorateAsJava {
     }
   }
 
-  trait Pathish[Rep] {
+  trait Metadataish {
+    def exists: Boolean
+    def path: Path
+    def allBytes: Array[Byte]
+    def nodeType: fs.NodeType
+    def size: Long
+    def permissions: PosixFilePermissions
+    def atime: Long
+    def mtime: Long
+
+    def name : String    = path.getFileName.to_s
+    def blockCount: Long = (size + blockSize - 1) / blockSize
+    def blockSize: Long  = 512 // FIXME
+  }
+
+  // this could have been an apply method in an object allowing the use of Metadata instead of Metadataish
+  // I however could not get myself to repeat all those arguments
```

Comment:
While I totally agree with everything you are saying. I do not think this pull request warrants the refactoring you propose.

I could make an auxiliary constructor with perms and times, but that kinda defeats the purpose of having a case class to implement the interface that is already implemented by `Pathish`.

Please know that I'm not sure how long I can keep myself from refactoring the whole thing. If I would, I rather do it in a separate pull request that does not introduce new features.